hilti/toolchain/src/ast/operators/string.cc

```diff
@@ -126,6 +126,68 @@ class Encode : public BuiltInMemberCall {
 };
 HILTI_OPERATOR_IMPLEMENTATION(Encode);
 
+class Split : public BuiltInMemberCall {
+public:
+    Signature signature(Builder* builder) const final {
+        return Signature{
+            .kind = Kind::MemberCall,
+            .self = {parameter::Kind::In, builder->typeString()},
+            .member = "split",
+            .param0 =
+                {
+                    .name = "sep",
+                    .type = {parameter::Kind::In, builder->typeString()},
+                    .optional = true,
+                },
```

Comment:
I think it's reasonable to just use a default, but one useful part that would get lost with this suggestion is that the no-argument `split()` will use any whitespace as a separator:

https://github.com/zeek/spicy/blob/3a50444e255e07cb47238eae88efa05f8c969f14/hilti/runtime/src/util.cc#L135

https://github.com/zeek/spicy/blob/3a50444e255e07cb47238eae88efa05f8c969f14/hilti/runtime/include/util.h#L159

Whereas `split(" ")` would just split on spaces, and `split(detail::whitespace_chars)` would not split on the individual whitespaces:

https://github.com/zeek/spicy/blob/3a50444e255e07cb47238eae88efa05f8c969f14/hilti/runtime/src/util.cc#L115

Instead, how about `split` called with an empty string (`""`) `delim` just switches its behavior to the old `split()` behavior (since I don't think the empty string argument would be useful anyway) and then the default is `.default_ = builder->stringLiteral("")` (not `" "`)?

Then that would also change for `bytes` and fix all of the documentation confusion I pointed out, from what I can tell, so long as the empty string case is documented properly

Not too sure if that's what you intended, but the code suggestion has `" "` as the default so just making sure that seems reasonable

Reply:
Yeah, you got me convinced :)